From 91b9050a908d74518c9945e12bb3e1318b3c1544 Mon Sep 17 00:00:00 2001
From: Bugdebugger <bugdebugger@users.noreply.github.com>
Date: Tue, 31 Mar 2015 01:42:54 +0200
Subject: [PATCH 086/129] Fix build when wiimote is enabled

Old logging calls changed to new logmich format
Remove old debug file reference
Fixed scons not finding boost/signals2.hpp

Fixes Pingus/pingus#146
---
 SConscript                                  |  2 +-
 src/engine/input/wiimote/wiimote_driver.cpp | 10 ++--------
 2 files changed, 3 insertions(+), 9 deletions(-)

diff --git a/SConscript b/SConscript
index 7d160fb28..d6c5b3e07 100644
--- a/SConscript
+++ b/SConscript
@@ -187,7 +187,7 @@ class Project:
                                                      'src/engine/input/xinput/xinput_device.cpp'])
 
     def configure_boost(self):
-        if not self.conf.CheckHeader('boost/signals2.hpp', '<>', 'c++'):
+        if not self.conf.CheckLibWithHeader('boost_signals', 'boost/signals2.hpp', 'c++'):
             self.fatal_error += "  * library 'boost_signals2' not found\n"
 
     def configure_png(self):
diff --git a/src/engine/input/wiimote/wiimote_driver.cpp b/src/engine/input/wiimote/wiimote_driver.cpp
index 4beb29221..191077ad8 100644
--- a/src/engine/input/wiimote/wiimote_driver.cpp
+++ b/src/engine/input/wiimote/wiimote_driver.cpp
@@ -17,7 +17,6 @@
 #include "engine/input/wiimote/wiimote_driver.hpp"
 
 #include "engine/input/wiimote/wiimote.hpp"
-#include "pingus/debug.hpp"
 #include "pingus/globals.hpp"
 #include "util/string_util.hpp"
 
@@ -48,9 +47,7 @@ WiimoteDriver::update(float delta)
     WiimoteEvent& event = *i;
     if (event.type == WiimoteEvent::WIIMOTE_BUTTON_EVENT)
     {
-      pout(PINGUS_DEBUG_INPUT) << "WiimoteDriver: (wiimote:button (button "
-                               << event.button.button << ")) => "
-                               << event.button.down << std::endl;
+      log_debug("WiimoteDriver: (wiimote:button (button %1)) => %2", event.button.button, event.button.down);
 
       for (std::vector<ButtonBinding>::const_iterator j = button_bindings.begin();
            j != button_bindings.end();
@@ -64,10 +61,7 @@ WiimoteDriver::update(float delta)
     }
     else if (event.type == WiimoteEvent::WIIMOTE_AXIS_EVENT)
     {
-      pout(PINGUS_DEBUG_INPUT) << "WiimoteDriver: (wiimote:axis (axis "
-                               << event.axis.axis << ")) => "
-                               << event.axis.pos
-                               << std::endl; // Fixme: should output string
+      log_debug("WiimoteDriver: (wiimote:axis (axis %1)) => %2", event.axis.axis, event.axis.pos); // Fixme: should output string
 
       for (std::vector<AxisBinding>::const_iterator j = axis_bindings.begin();
            j != axis_bindings.end(); ++j)
-- 
2.51.2

